### PR TITLE
Sneks

### PR DESCRIPTION
**Pull request by Pacmandevil**
Adds a subtype of cat: Snakes.
There's really not much to add to this they're pretty similar. Also adds Corpus because bones is not the GCAT runtime was, and I feel Runtime should be IT's pet. Not actually mapped in though.

This is a flat port from VG.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 1]
```diff
@@ -249,6 +249,68 @@
...
+		if (friend.stat >= DEAD || friend.health <= config.health_threshold_softcrit)
+			if (prob((friend.stat < DEAD)? 50 : 15))
+				var/verb = pick("Hisses")
+				audible_emote(pick("[verb] in distress.", "[verb] anxiously."))
+		else
+			if (prob(5))
+				visible_emote(pick("nuzzles [friend].",
+								   "brushes against [friend].",
+								   "rubs against [friend].",
+								   "Hisses seductively."),0)
```
Comment (Lohikar) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: :thinking:

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 2]
```diff
@@ -249,6 +249,68 @@
...
+	else if (friend.health <= 50)
+		if (prob(10))
+			var/verb = pick("Hisses")
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: LLiterally haram. (`verb` is a reserved word. Do not use it as a variable name.)

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 3]
```diff
@@ -249,6 +249,68 @@
...
+	friend = usr
+
+/mob/living/simple_animal/cat/snek/fluff/Life()
+	..()
+	if (stat || QDELETED(friend))
+		return
+	if (get_dist(src, friend) <= 1)
+		if (friend.stat >= DEAD || friend.health <= config.health_threshold_softcrit)
+			if (prob((friend.stat < DEAD)? 50 : 15))
+				var/verb = pick("Hisses")
```
Comment (Lohikar) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: Might as well just hard-code this if there's only one element.

File: code/modules/mob/living/simple_animal/friendly/cat.dm (hunk 2 above)
Comment (Lohikar) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: Same here.

File: code/modules/mob/living/simple_animal/friendly/cat.dm (hunk 3 above)
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: Same: `verb` == reserved.

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 4]
```diff
@@ -249,6 +249,68 @@
 	befriend_job = "Chief Medical Officer"
 	holder_type = /obj/item/weapon/holder/cat/black
 
+//Advanced Noodle AI.
+/mob/living/simple_animal/cat/snek/fluff
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: This smells like a meme.

Comment (Pacmandevil) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: structured the same way as the runtime is.

**Comment by Alberyk** [comment]
Why is this bypassing dev?

**Comment by SoundScopes** [comment]
What's the reason for snakes being sub-typed under cat rather than being it's own type

**Comment by Pacmandevil** [comment]
@SoundScopes 
So they chase and eat mice.

**Comment by SoundScopes** [comment]
If that's the only reason then there has to be a better way of doing it.
The snakes Life() will call the cats Life() which as a tiny probability of it "mrowls fiercely"

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 5]
```diff
@@ -249,6 +249,66 @@
...
+//something you shouldn't step on
+/mob/living/simple_animal/cat/snek
+	name = "snake"
+	desc = "sssSSSSsss"
+	icon_state = "snek"
+	icon_living = "snek"
+	icon_dead = "snek_dead"
+	gender = NEUTER
+	speak = list("SssssSSSS.", "Slirp.","HSSSSS")
+	speak_emote = ("hisses")
```
Comment (Lohikar) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: This should be a list, no?

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 6]
```diff
@@ -249,6 +249,66 @@
...
+	emote_hear = ("hisses")
```
Comment (Lohikar) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: This too.

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 7]
```diff
@@ -249,6 +249,66 @@
...
+	emote_see = list("slithers, tastes the air.")
```
Comment (Lohikar) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: This meant to be one string?

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 8]
```diff
@@ -249,6 +249,66 @@
...
+				audible_emote(pick("hisses."))
+		else
+			if (prob(5))
+				visible_emote(pick("nuzzles [friend].",
+								   "brushes against [friend].",
+								   "rubs against [friend].",
+								   "hisses happily."),0)
+	else if (friend.health <= 50)
+		if (prob(10))
+			audible_emote("hisses.")
```
Comment (Lohikar) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: I feel like this should be different from the random speak message.

**Comment by Pacmandevil** [comment]
Don't feel there's actually a reason to add this.

**Pull request closed without merging**